docs/minio-limits.md

```diff
@@ -28,6 +28,9 @@
 | Maximum number of parts returned per list parts request                         | 10000                                         |
 | Maximum number of objects returned per list objects request                     | 4500                                          |
 | Maximum number of multipart uploads returned per list multipart uploads request | 1000                                          |
+| Maximum length for bucket names                                                 | 63                                            |
+| Maximum length for object names                                                 | 1024                                          |
+| Maximum length for '/' seperated object name segment                            | 255                                           |
```

Comment:
@harshavardhana this is to say that:

`/bucket/segment1/segment2/segment3`

each `segment` must be < 255 chars long? Just checking - when I import these docs I will probably need to add an example here.

Also, for object names, does this include the slash delimiters? I assume so, but am thinking whether it starts at `/object-name` or `object`

Reply:
Yes this is correct

Reply:
Good to know. I need to import these limits into next-gen docs, will see about adding examples later if necessary.

Reply:
@ravindk89 The only thing we're not 100% sure about is, if the 255 chars per segment is an official AWS limitation or a Minio specific one.
Where 1024 chars for object names and 63 chars for bucket names are official AWS limitations that also applies to Minio.

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html - 3 -> 63 character bucket name limit

https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html - object key max name of 1024 bytes

These look like AWS limits. For the object name segment, I'll defer to @harshavardhana - I would venture this is a limit we encountered through reverse engineering the AWS API behavior, vs something AWS explicitly documents. But I could be wrong, so have asked for clarity.

Reply:
The reason for segments on MinIO is because of the filesystem as namespace requirement for MinIO. We are database-free object storage, unlike every other vendor.

Reply:
OK - that implies that out of those limits, only the segment limit is MinIO-specific. The other two are AWS inherited. @harshavardhana

Reply:
> OK - that implies that out of those limits, only the segment limit is MinIO-specific. The other two are AWS inherited. @harshavardhana 

Correct @ravind89